astropy/coordinates/representation.py

```diff
@@ -483,7 +554,141 @@ def represent_as(self, other_class):
+
+                new_rep = other_class[0].from_cartesian(self.to_cartesian())
+
+                new_diffs = []
+                for i,diff in enumerate(self.differentials):
+                    new_diffs.append(diff.represent_as(other_class[1+i],
+                                                       base=self))
+                new_rep._differentials = tuple(new_diffs)
+
+            else:
+                new_rep = other_class.from_cartesian(self.to_cartesian())
+
+            return new_rep
+
+    def with_differentials(self, differentials):
+        """
+        Create a new representation with the same positions as this
+        representation, but with new differentials.
+
+        Parameters
+        ----------
+        differentials : Sequence of `~astropy.coordinates.BaseDifferential`
+            The differentials for the new representation to have.
+
+        Returns
+        -------
+        newrepr
+            A copy of this representation, but with the ``differentials`` as
+            its differentials.
+        """
+        # Implemented this way for (1) performance (skips going through the
+        # initializer), and (2) will preserve any additional attributes a user
+        # may have set
+
+        new_differentials = self._validate_differentials(differentials)
+
+        # save the differentials, then strip them so that the copy doesn't do
+        # an unneeded copy, then re-attach them to the right objects
+        olddiffs = self._differentials
+        try:
+            self._differentials = None
+            newrepr = deepcopy(self)
+            newrepr._differentials = new_differentials
+            return newrepr
+        finally:
+            self._differentials = olddiffs
+
+    def without_differentials(self):
+        """
+        Create a new representation with the same positions as this
+        representation, but without any differentials.
+
+        Returns
+        -------
+        newrepr
+            A copy of this representation, but without the ``differentials``.
+        """
+
+        # Again, implemented this way for (1) performance (skips going through
+        # the initializer), and (2) will preserve any additional attributes a
+        # user may have set
+
+        # save the differentials, then strip them so that the copy doesn't do
+        # an unneeded copy, then re-attach them to the right objects
+        olddiffs = self._differentials
+        try:
+            self._differentials = None
+            newrepr = deepcopy(self)
+            newrepr._differentials = ()
+            return newrepr
+        finally:
+            self._differentials = olddiffs
+
+    def to_cartesian(self, diffstocart=False):
+        """
+        Convert the representation to its carteisan form.
+
+        Note that subclasses should *not* override this - rather they should
+        override the ``_to_cartesian_helper()`` method.
+
+        Parameters
+        ----------
+        diffstocart : bool
```

Comment:
This would disappear if we make `BaseRepresentation.represent_as(<BaseRepresentation>)` drop differentials. Alternatively, we'd have to add `without_differentials()` where appropriate. Will think a bit more.